service/matching/taskReader.go

```diff
@@ -361,7 +367,15 @@ func (tr *taskReader) completeTask(task *persistence.TaskInfo, err error) {
 
 func (tr *taskReader) newDispatchContext(isolationGroup string) (context.Context, context.CancelFunc) {
 	if isolationGroup != "" {
-		return context.WithTimeout(tr.cancelCtx, tr.config.AsyncTaskDispatchTimeout())
+		domainEntry, err := tr.domainCache.GetDomainByID(tr.taskListID.domainID)
+		if err != nil {
+			// we don't know if the domain is active in the current cluster, assume it is active and set the timeout
+			return context.WithTimeout(tr.cancelCtx, tr.config.AsyncTaskDispatchTimeout())
```

Comment:
why is the timeout only set on the active side?

Reply:
because the passive side is not actually polling

Reply:
If it isn't, why do we care about differentiating it then?